block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagRepresentationSyntax.scala

```diff
@@ -12,60 +14,67 @@ import coop.rchain.models.syntax._
 import coop.rchain.shared.syntax._
 import fs2.Stream
 
-trait BlockDagRepresentationSyntax {
-  implicit final def blockStorageSyntaxBlockDagRepresentation[F[_]](
-      dag: BlockDagRepresentation[F]
-  ): BlockDagRepresentationOps[F] = new BlockDagRepresentationOps[F](dag)
+trait DagRepresentationSyntax {
+  implicit final def blockStorageSyntaxDagRepresentation[F[_]](
+      dag: DagRepresentation
+  ): DagRepresentationOps[F] = new DagRepresentationOps[F](dag)
 }
 
 final case class BlockDagInconsistencyError(message: String) extends Exception(message)
 final case class NoLatestMessage(message: String)            extends Exception(message)
 
-final class BlockDagRepresentationOps[F[_]](
-    // BlockDagRepresentation extensions / syntax
-    private val dag: BlockDagRepresentation[F]
+final class DagRepresentationOps[F[_]](
+    // DagRepresentation extensions / syntax
+    private val dag: DagRepresentation
 ) extends AnyVal {
 
-  /**
-    * Get block metadata, "unsafe" because method expects block already in the DAG.
-    *
-    * Unfortunately there is no way to get stack trace when error is thrown in async execution.
-    * Monix does not have support and cats.effect support is in creation.
-    * https://github.com/typelevel/cats-effect/pull/854
-    * So extra source parameters are a desperate measure to indicate who is the caller.
-    */
-  def lookupUnsafe(hash: BlockHash)(
-      implicit sync: Sync[F]
-  ): F[BlockMetadata] = {
-    def errMsg = s"DAG storage is missing hash ${PrettyPrinter.buildString(hash)}"
-    dag.lookup(hash) >>= (_.liftTo(BlockDagInconsistencyError(errMsg)))
-  }
+  def lastFinalizedBlockUnsafe(implicit s: Sync[F]): F[BlockHash] =
+    DagRepresentation.lastFinalizedBlockUnsafe(dag)
 
-  def lookupUnsafe(
-      hashes: Seq[BlockHash]
-  )(implicit concurrent: Concurrent[F]): F[List[BlockMetadata]] = {
-    val streams = hashes.map(h => fs2.Stream.eval(lookupUnsafe(h)))
-    fs2.Stream.emits(streams).parJoinUnbounded.compile.toList
-  }
+  def invalidBlocks(implicit s: Sync[F], bds: BlockDagStorage[F]): F[Set[BlockMetadata]] =
+    DagRepresentation.invalidBlocks(dag)
+
+  def latestMessageHash(
+      validator: Validator
+  )(implicit s: Sync[F], bds: BlockDagStorage[F]): F[Option[BlockHash]] =
+    DagRepresentation.latestMessageHash(dag, validator)
+
+  def latestMessageHashes(
+      implicit s: Sync[F],
+      bds: BlockDagStorage[F]
```

Comment:
Interestingly if you add `Sync[F]` implicit at the end you can avoid `implicit s: Sync[Task]` definition in tests. It seems that  block storage will carry information for F and be visible to Sync.